src/Microsoft.InsightsGenerator/DataTransformation.cs

```diff
@@ -78,29 +78,29 @@ private string[] GetColumnLabels(DataArray array, out DataArray.DataType[] colum
             {
                 if (columnInfo.ContainsKey(DataArray.DataType.String))
                 {
-                    int minDistinctValue = -1;
-                    int minColumnIndex = -1;
-
+                    int maxDistinctValue = Int16.MaxValue;
+                    int maxColumnIndex = -1;
+                    int maxColumnLabelIndex = 0;
                     List<ColumnInfo> stringColumns = columnInfo[DataArray.DataType.String];
                     for (int i = 0; i < stringColumns.Count; ++i)
                     {
-                        if (minDistinctValue == -1 || minDistinctValue > stringColumns[i].DistinctValues)
+                        if (maxDistinctValue == Int16.MaxValue || maxDistinctValue < stringColumns[i].DistinctValues)
                         {
-                            minDistinctValue = stringColumns[i].DistinctValues;
-                            minColumnIndex = i;
+                            maxDistinctValue = stringColumns[i].DistinctValues;
+                            maxColumnIndex = i;
+                            maxColumnLabelIndex = stringColumns[i].ColumnIndex;
                         }
                     }
                     
-                    labels[stringColumns[minColumnIndex].ColumnIndex] = "input_g_0";
-
+                    labels[maxColumnLabelIndex] = "input_g_0";
```

Comment:
I thought the expected logic is that the string column with the least amount of distinct values should be the grouping column.  Why are we making this change?